src/main/java/ch/njol/skript/effects/EffWhile.java

```diff
@@ -0,0 +1,95 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.TriggerItem;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Inline While")
+@Description("A shorthand effect for executing an effect while a condition is met.")
+@Examples({"on join:",
+	"\tadd (random integer between 1 and 5) to {_amount} while chance of 50%",
+	"\tgive {_amount} of dirt to player"})
+@Since("INSERT VERSION")
+public class EffWhile extends Effect {
+	
+	static {
+		Skript.registerEffect(EffWhile.class, "[(1:do)] <.+> while <.+>");
```

Comment:
Probably will drop support for `do while` since it's very confusing.

Reply:
I believe keeping `do while` is a better thing as there's times I'd prefer to make sure I always add 1 to stuff this is better than  doing it once before the while